### PR TITLE
Add docstring to the insertion method & add empty result note

### LINKED ISSUE #21364: pandas read_sql missing dtype on empty result
#### Code Sample, a copy-pastable example if possible

con = some sqlalchemy engine
query = "select * from table"
result = pd.read_sql(query, con)

#### Problem description

Assume table have two columns, user (str) and id (int).
Previously, even if the result has 0 rows, the columns would have dtype in consistent with the table column.

result.dtypes
user    object
**id       int64**
dtype: object

In pandas 0.23, if the result is empty, then all the columns dtypes in result are object.

result.dtypes
user    object
**id      object**
dtype: object

#### Expected Output
user    object
id       int64
dtype: object

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.5.final.0
python-bits: 64
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 62 Stepping 4, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.0
pytest: 3.5.1
pip: 10.0.1
setuptools: 39.1.0
Cython: 0.28.2
numpy: 1.14.3
scipy: 1.1.0
pyarrow: 0.9.0
xarray: 0.10.4
IPython: 6.4.0
sphinx: None
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.4
blosc: None
bottleneck: 1.2.1
tables: None
numexpr: 2.6.5
feather: 0.4.0
matplotlib: 2.2.2
openpyxl: 2.5.3
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.4
lxml: 4.2.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.7
pymysql: 0.8.1
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### LINKED ISSUE #26868: DOC: Reference to type 'pd_table' in IO Tools User Guide
Insertion Method section of the reference refers to "pd_table" in the function signature. This is the only place it is used in the whole documentation. Search engines results for "pd_table" are mostly examples where it's used to refer to pd.DataFrame, so it's rather confusing.

Not sure what's the best way to make this clear but the signature should refer to `pandas.io.SQLTable()`.

This was added in #21401. I came across this while working on #26795.

### PR DESCRIPTION
**Pull request by oguzhanogreden**
closes #21364 
closes #26868 
- [x] tests passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- **What's new?** Some small fixes in documentation.

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -5012,6 +5012,18 @@ Example of a callable using PostgreSQL `COPY clause
...
+      """
+      Execute SQL statement inserting data
+
+      Parameters
+      ----------
+      table : pandas.io.sql.SQLTable
+      conn : sqlalchemy.engine.Engine or sqlalchemy.engine.Connection
+      keys : list of str
+          Column names
+      data_iter : generator of list
```
Comment (jreback) [inline doc/source/user_guide/io.rst]: -> Iterable

File: doc/source/user_guide/io.rst [hunk 2]
```diff
@@ -5045,6 +5057,16 @@ table name and optionally a subset of columns to read.
 
    pd.read_sql_table('data', engine)
 
+Note that pandas infers column dtypes from query outputs, and not by looking
```
Comment (jreback) [inline doc/source/user_guide/io.rst]: you can make this a note section

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by oguzhanogreden** [comment]
Addressed the comments and updated with the current master.

File: doc/source/user_guide/io.rst [hunk 3]
```diff
@@ -5080,6 +5091,18 @@ table name and optionally a subset of columns to read.
 
    pd.read_sql_table('data', engine)
 
+.. note::
+
+  Note that pandas infers column dtypes from query outputs, and not by looking
```
Comment (TomAugspurger) [inline doc/source/user_guide/io.rst]: Does this need to be indented to be under the `n` in note`?

Comment (oguzhanogreden) [reply doc/source/user_guide/io.rst]: It seems not, see for instance the note at line 4884, which is rendered correctly. In general, indentations for the note sections are inconsistent in this file. If there's anything in the guidelines, I missed it (twice now :)),

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**